File: code/network_evaluation.R
library(igraph, lib.loc = "../software/deps_R")
#library(igraph)

data <- read.csv("../results/network.csv")

my_network <- graph.data.frame(data, directed = FALSE)

svg("../report/figures/network_igraph.svg")
# Define un vector de colores para los nodos
# Inicialmente todos los nodos son de color "yellow"
colores_nodos <- rep("yellow", vcount(my_network))  
# Cambia el color del nodo con el nombre "GNB3" a rojo
colores_nodos[V(my_network)$name == "GNB3"] <- "red"  
plot(my_network,
     vertex.label.cex = 0.5,
     vertex.label.color = "black",
     vertex.size = 12,
     vertex.color = colores_nodos,
     layout = layout_nicely)
dev.off()

# Calcumos las comunidades mediante el método "cluster_edge_betweenness"
ceb <- cluster_edge_betweenness(my_network)

svg("../report/figures/network_comunidades.svg")
plot(ceb, my_network,
     vertex.label.cex = 0.5,
     vertex.label.color = "black",
     vertex.size = 12,
     layout = layout_nicely)
dev.off()

# Asignamos a cada nodo su comunidad
comunidades <- membership(ceb)

# Nos quedamos con los nodos que están en la misma comunidad
# que nuestro en de interés
comunidad_interes <- which(comunidades == 2)

# Nos quedamos con los nombres de los nodos
nombres_interes <- V(my_network)$name[comunidad_interes]

# Guarda los nombres de los nodos en un archivo CSV
write.csv(data.frame(nombres_interes),
          file = "../results/interest_nodes.csv",
          row.names = FALSE, quote = FALSE)

File: code/network_evaluation.py
from igraph import Graph, plot
import sys
import pandas as pd
import os

"""
Calcula el grafo de una red y el grafo de sus comunidades.
Guarda el nombre de los nodos de la comunidad donde se encuentra el gen
de interés.
"""

# Obtener la ruta absoluta del script actual
path = os.path.dirname(os.path.abspath(__file__))

# Construir la ruta completa al archivo CSV
csv_path = os.path.join(path, '..', 'results', sys.argv[1])

# Cargar datos desde el archivo CSV
data = pd.read_csv(csv_path)

g = Graph.TupleList(data.itertuples(index=False),directed=False)
g = g.simplify()

colors = ["lightgreen" if name == "GNB3" else "#FF6961" for name in g.vs["name"]]

# Guardar el grafo en formato SVG
plot(g, os.path.join(path, '..', 'results', sys.argv[2]), vertex_label=g.vs["name"], vertex_label_size=10, vertex_label_color="black",
     vertex_size=12, vertex_label_dist=2, edge_color="darkgray", vertex_color=colors)

# Calcular las comunidades mediante el método "cluster_edge_betweenness"
ceb = g.community_edge_betweenness()
cl = ceb.as_clustering()

# Guardar el grafo de comunidades en formato SVG
plot(cl, os.path.join(path, '..', 'results', sys.argv[3]), mark_groups=True, vertex_label=g.vs["name"], vertex_label_size=10, vertex_label_color="black",
     vertex_size=12, vertex_label_dist=2, edge_color="darkgray", vertex_color="#FF6961")

# Obtener los nodos en la misma comunidad que nuestro nodo de interés
for comunidad in (cl):
    nombres = g.vs[comunidad]["name"]
    if "GNB3" in nombres:
        # Guardar los nombres de los nodos en un archivo CSV
        nombres_df = pd.DataFrame({'Nombres': nombres})
        csv_path = os.path.join(path, '..', 'results', sys.argv[4])
        nombres_df.to_csv(csv_path, index=False, header=False)
        break
